todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/TasksRepository.java

```diff
@@ -234,78 +146,25 @@ public void clearCompletedTasks() {
     @Override
     public Observable<Task> getTask(@NonNull final String taskId) {
         checkNotNull(taskId);
-
-        final Task cachedTask = getTaskWithId(taskId);
-
-        // Respond immediately with cache if available
-        if (cachedTask != null) {
-            return Observable.just(cachedTask);
-        }
-
-        // Load from server/persisted if needed.
-
-        // Do in memory cache update to keep the app UI up to date
-        if (mCachedTasks == null) {
-            mCachedTasks = new LinkedHashMap<>();
-        }
-
-        // Is the task in the local data source? If not, query the network.
-        Observable<Task> localTask = getTaskWithIdFromLocalRepository(taskId);
-        Observable<Task> remoteTask = mTasksRemoteDataSource
-                .getTask(taskId)
-                .doOnNext(task -> {
-                    mTasksLocalDataSource.saveTask(task);
-                    mCachedTasks.put(task.getId(), task);
-                });
-
-        return Observable.concat(localTask, remoteTask).first()
-                .map(task -> {
-                    if (task == null) {
-                        throw new NoSuchElementException("No task found with taskId " + taskId);
-                    }
-                    return task;
-                });
+        return mTasksLocalDataSource.getTask(taskId);
     }
 
     @Override
     public void refreshTasks() {
-        mCacheIsDirty = true;
+        mTasksRemoteDataSource.getTasks()
+                .subscribeOn(Schedulers.io())
```

Comment:
In the rxjava samples, we've been using a `BaseSchedulerProvider` for scheduling, and then that class gets to be mocked in tests. Should we have a consistent approach?